roxie/ccd/ccdactivities.cpp

```diff
@@ -3075,7 +3075,7 @@ class CRoxieIndexActivityFactory : public CRoxieKeyedActivityFactory
         {
             bool isOpt = (helper->getFlags() & TIRoptional) != 0;
             OwnedRoxieString indexName(helper->getFileName());
-            datafile.setown(queryFactory.queryPackage().lookupFileName(indexName, isOpt, true, true, queryFactory.queryWorkUnit()));
+            datafile.setown(queryFactory.queryPackage().lookupFileName(indexName, isOpt, true, true, queryFactory.queryWorkUnit(), !queryFactory.isDynamic()));
```

Comment:
Would be clearer to pass in true (we have already tested queryFactory.isDynamic() above - if we are in this branch then we know we are doing a pre-resolve of the files, which implies that foreign should be ignored.